### DISCUSSION TITLE
Software reset to bootloader mode - code example

### DISCUSSION THREAD

**Discussion by madias123** [earlephilhower/arduino-pico#1512] [Ideas]
Maybe worth as an example:
A simple code, triggering the Pi Pico into bootloader upload mode. (You'll need this for zero-knowledge customers, so they can update the firmware simply with drag&drop when the select "update firmware"):

```
extern "C" {
#include "pico.h"
#include "pico/time.h"
#include "pico/bootrom.h"
}
```
Then force upload mode with just one line:
`reset_usb_boot(1<<PICO_DEFAULT_LED_PIN,0); //invokes reset into bootloader mode`

Credits: Found this here:
[https://forum.arduino.cc/t/programming-pi-pico-from-host-getting-into-bootmode-without-user-interaction/956170](url)

**Comment by earlephilhower** [comment]
Why not add it to `rp2040` as `rp2040.rebootToBootloader()` ?

I.e. extend: https://github.com/earlephilhower/arduino-pico/blob/3f475ac68cb3d89cd934cf110b81e90ffabff1da/cores/rp2040/RP2040Support.h#L309

Want to do a PR?
